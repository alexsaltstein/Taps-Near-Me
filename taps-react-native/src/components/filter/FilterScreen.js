import React from 'react';
import { View, Text, StyleSheet, TextInput } from 'react-native';
import { COLORS } from '../../styles/COLORS';
import BackButton from '../widgets/BackButton';
import ImageInput from './ImageInput';
import NumberInput from './NumberInput';
import SliderInput from './SliderInput';
import setStatusBarColor from '../utils/StatusBarColorFunctions';
import TypeInput from './TypeInput';

const FilterScreen = ({ navigation }) => {
  const [setColor] = setStatusBarColor();

  const [type, setType] = React.useState('');

  const [radius, setRadius] = React.useState(0);

  const [rating, setRating] = React.useState(-1);

  const [maxABVIndex, setMaxABVIndex] = React.useState(-1);
  const [minABVIndex, setMinABVIndex] = React.useState(-1);

  const [maxIndex, setMaxIndex] = React.useState(-1);
  const [minIndex, setMinIndex] = React.useState(-1);


  React.useEffect(() => {
    navigation.addListener('focus', () => {
      setColor(COLORS.purple);
    })
  }, [])

  return (
    <View style={styles.container}>
      <BackButton navigation={navigation} color={COLORS.white} />
      <Text style={styles.title}>Map Filter</Text>
      <View style={styles.formContainer}>
        <TypeInput
          label="Type:"
          curr={type}
          setCurr={setType} />
        <SliderInput
          label="Radius"
          curr={radius}
          setCurr={setRadius} />
        <ImageInput
          label="Minimum Rating:"
          curr={rating}
          setCurr={setRating} />
        <NumberInput
          minIndex={minABVIndex}
          setMinIndex={setMinABVIndex}
          maxIndex={maxABVIndex}
          setMaxIndex={setMaxABVIndex}
          vals={['<4', '5', '6', '7', '8>']}
          label="abv" />
        <NumberInput
          minIndex={minIndex}
          setMinIndex={setMinIndex}
          maxIndex={maxIndex}
          setMaxIndex={setMaxIndex}
          vals={['<20', '30', '40', '50', '60>']}
          label="ibu" />
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.purple,
  },
  title: {
    fontFamily: 'open-sans-semi',
    color: COLORS.white,
    fontSize: 30,
    alignSelf: 'center'
  },
  formContainer: {
    paddingHorizontal: 20,
    paddingVertical: 10,
  },
})

export default FilterScreen;