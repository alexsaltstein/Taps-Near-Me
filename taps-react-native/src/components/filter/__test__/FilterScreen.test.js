import React from 'react';
import renderer from 'react-test-renderer';
import FilterScreen from '../FilterScreen';
import ReactDOM from 'react-dom';

test('renders correctly', () => {
  const tree = renderer.create(<FilterScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});