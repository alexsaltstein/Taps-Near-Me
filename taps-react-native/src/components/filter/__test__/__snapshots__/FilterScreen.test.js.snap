// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-flex-13awgt0 r-justifyContent-1777fci"
>
  <div
    className="css-view-1dbjc4n r-alignSelf-k200y r-cursor-1loqt21 r-justifyContent-1777fci r-left-1t68eob r-position-u8s1d r-top-12pp1o0 r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
    data-focusable={true}
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "WebkitTransitionDuration": "0s",
        "transitionDuration": "0s",
      }
    }
    tabIndex="0"
  >
    <div
      className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
      style={
        Object {
          "height": "30px",
          "width": "30px",
        }
      }
    >
      <div
        className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
        style={
          Object {
            "WebkitFilter": "url(#tint-0)",
            "filter": "url(#tint-0)",
          }
        }
      />
      <svg
        style={
          Object {
            "height": 0,
            "position": "absolute",
            "visibility": "hidden",
            "width": 0,
          }
        }
      >
        <defs>
          <filter
            id="tint-0"
            suppressHydrationWarning={true}
          >
            <feFlood
              floodColor="#000000"
            />
            <feComposite
              in2="SourceAlpha"
              operator="atop"
            />
          </filter>
        </defs>
      </svg>
    </div>
  </div>
  <div
    className="css-text-901oao"
    dir="auto"
    onClick={[Function]}
  >
    Filter Screen
  </div>
</div>
`;
